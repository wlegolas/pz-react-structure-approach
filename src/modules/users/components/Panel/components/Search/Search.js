import PropTypes from 'prop-types'
import React from 'react'

const Search = ({ onKeyUp }) => {
  return (
    <div className="pz-search">
      <input
        type="text"
        placeholder="Fill the github user and press enter to search"
        onKeyUp={onKeyUp}
      />
    </div>
  )
}

Search.propTypes = {
  onKeyUp: PropTypes.func.isRequired
}

export default Search
