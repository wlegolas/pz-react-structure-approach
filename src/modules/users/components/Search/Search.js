import PropTypes from 'prop-types'
import React from 'react'

class Search extends React.Component {
  static propTypes = {
    onSearch: PropTypes.func.isRequired
  }

  handleClick = event => {
    console.log('***** Foi')
  }

  handleKeyUp = event => {
    const keyCode = event.which || event.keyCode
    const ENTER = 13

    if (keyCode === ENTER) {
      this.props.onSearch(event.target.value)
    }
  }

  render() {
    return (
      <div className="pz-search">
        <input
          type="text"
          placeholder="Fill the github user and press enter to search"
          onKeyUp={this.handleKeyUp}
        />
        <button type="button" onClick={this.handleClick}>
          Teste
        </button>
      </div>
    )
  }
}

export default Search
