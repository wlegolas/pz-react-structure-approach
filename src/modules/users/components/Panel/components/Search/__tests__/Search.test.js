import chai from 'chai'
import toJson from 'enzyme-to-json'
import React from 'react'
import Search from '../Search'

describe('<Search />', () => {
  let wrapper

  beforeAll(() => {
    const handleKeyUp = jest.fn()
    wrapper = shallow(<Search onKeyUp={handleKeyUp} />)
  })

  describe('render()', () => {
    it('should render the component', () => {
      expect(toJson(wrapper)).to.matchSnapshot()
    })
  })

  describe('verify the structure', () => {
    it('should have a container with the class "pz-search"', () => {
      expect(wrapper.find('div.pz-search')).to.have.lengthOf(1)
    })

    it('should have an input element', () => {
      expect(wrapper.find('input')).to.have.lengthOf(1)
    })

    it('should have an input with the attribute "placeholder" defined', () => {
      expect(wrapper.find('input').prop('placeholder')).to.be.equal(
        'Fill the github user and press enter to search'
      )
    })
  })

  describe('onKeyUp', () => {
    it('should call the function when event "keyup" is fired', () => {
      const handleKeyUp = chai.spy()
      wrapper = shallow(<Search onKeyUp={handleKeyUp} />)

      wrapper.find('input').simulate('keyup')

      expect(handleKeyUp).to.have.been.called.exactly(1)
    })

    it('should not call the functions when event "keydown" is fired', () => {
      const handleKeDown = chai.spy()
      wrapper = shallow(<Search onKeyUp={handleKeDown} />)

      wrapper.find('input').simulate('keydwon')

      expect(handleKeDown).to.have.been.called.exactly(0)
    })
  })
})
