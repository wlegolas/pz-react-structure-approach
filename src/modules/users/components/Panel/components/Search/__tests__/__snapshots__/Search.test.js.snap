// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> render() should render the component 1`] = `
<div
  className="pz-search"
>
  <input
    onKeyUp={[MockFunction]}
    placeholder="Fill the github user and press enter to search"
    type="text"
  />
</div>
`;
